### PR TITLE
Add write cache for access log

### PR DESCRIPTION
**Pull request by aoiasd**
/kind improvement

### CODE REVIEW

File: internal/proxy/accesslog/log_cache.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		bytes: make([]byte, 0),
+	}
+}
+
+func (c *writeCache) Cap() int {
+	return len(c.bytes)
+}
+
+func (c *writeCache) Full() bool {
+	return c.Cap() >= c.size
```
Comment (yah01) [inline internal/proxy/accesslog/log_cache.go:37]: in the context, `Cap` should be `Size` and `size` should be `cap`

File: internal/proxy/accesslog/log_cache.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslog
+
+type writeCache struct {
```
Comment (yah01) [inline internal/proxy/accesslog/log_cache.go:19]: in fact I think we don't need to write this by self, `bufio.Writer` may what you need

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/proxy/accesslog/log_cache.go (hunk 2 above)
Comment (yah01) [reply internal/proxy/accesslog/log_cache.go:19]: the impl doesn't guarantee the size of data never overs the cap set by config

Comment (aoiasd) [reply internal/proxy/accesslog/log_cache.go:19]: > the impl doesn't guarantee the size of data never overs the cap set by config

That just I need, access log won't write a very large data one time, allow over cache maxsize a littler for logger could close cache by set size to zero.

Comment (aoiasd) [reply internal/proxy/accesslog/log_cache.go:19]: > in fact I think we don't need to write this by self, `bufio.Writer` may be what you need

Now access logger was impl of zap.Logger, if use bufio.Writer we need split write method of Logger to a Writer, seems will need change more code.

File: internal/proxy/accesslog/access_log.go [hunk 3]
```diff
@@ -72,7 +73,12 @@ func InitAccessLogger(logCfg *paramtable.AccessLogConfig, minioCfg *paramtable.M
 			return nil, err
 		}
 
-		writeSyncer = zapcore.AddSync(lg)
+		if logCfg.CacheSize.GetAsInt() > 0 {
+			blg := bufio.NewWriterSize(lg, logCfg.CacheSize.GetAsInt())
```
Comment (yah01) [inline internal/proxy/accesslog/access_log.go:77]: `bufio` `Writer` is not thread-safe, you need to wrap it with a `RWMutex`

**Comment by aoiasd** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**